oauth/clientfactory/workspace_aware_client_factory.go

```diff
@@ -58,12 +58,14 @@ func (w WorkspaceAwareK8sClientFactory) CreateClient(ctx context.Context) (clien
 		lg.Error(err, "failed to fetch workspace name via API:", "api", w.RestConfig.Host, "workspace", wsName, "error", err.Error())
 		return nil, fmt.Errorf("unable to fetch workspace name via API: %w", err)
 	}
-	w.RestConfig.Host, err = url.JoinPath(w.RestConfig.Host, "workspaces", wsName)
+	// make local copy to avoid global config modification
+	restConfig := *w.RestConfig
```

Comment:
I would even do deepcopy, to be safe. wdyt ?

Reply:
yep let's do copy config